libs/rendering/sources/Key.cpp

```diff
@@ -5,46 +5,37 @@
 ** Key
 */
 
-#include "Key.hpp"
 #include "Components/Velocity.hpp"
+
+#include "Events/KeyDown.hpp"
+#include "Events/KeyPressed.hpp"
+#include "Key.hpp"
+
 #include "SparseArray.hpp"
+
 #include "raylib.h"
 
-rendering::system::Key::Key(
-    SparseArray<Component::Controllable> &controllables,
-    SparseArray<Component::Velocity> &velocities, std::size_t playerNumber
-)
-    : _controllables(controllables)
-    , _velocities(velocities)
-    , _playerNumber(playerNumber)
+constexpr KeyboardKey HANDLED_KEY[6]
+    = { KEY_W, KEY_Z, KEY_A, KEY_Q, KEY_S, KEY_D };
+
+rendering::system::Key::Key(event::EventQueue &events)
+    : _events(events)
 {
 }
 
 rendering::system::Key::~Key() = default;
 
 void rendering::system::Key::operator()()
 {
-    Component::Velocity velocity;
-    if (IsKeyDown(KEY_W) || IsKeyDown(KEY_Z)) {
-        velocity._vy -= 5.0f;
-    }
-    if (IsKeyDown(KEY_A) || IsKeyDown(KEY_Q)) {
-        velocity._vx -= 5.0f;
-    }
-    if (IsKeyDown(KEY_S)) {
-        velocity._vy += 5.0f;
-    }
-    if (IsKeyDown(KEY_D)) {
-        velocity._vx += 5.0f;
-    }
-
-    for (auto it = _velocities.begin(); it != _velocities.end(); ++it) {
-        auto &controllable = _controllables[it.get_entity()];
-        bool isControllable
-            = controllable && controllable->_id == _playerNumber;
+    // for (char keyPressed = GetKeyPressed(); keyPressed != 0;) {
+    //_events.addEvent<event::KeyPressed>(
+    // keyPressed
+    //);
+    //}
 
-        if (not isControllable)
-            continue;
-        **it = velocity;
+    for (int i = 0; i != 6; ++i) {
```

Comment:
au lieu de mettre un 6 en dur, tu peux utiliser un range-based for sur le tableau

Reply:
et enleve le code commenté